### PR TITLE
[random] add support for sets

### PR DESCRIPTION
**Pull request by Wh1teDuke**

### CODE REVIEW

**Comment by GULPF** [comment]
The other procs that return a random element from a collection are called `sample`, so this should use the same naming.

**Comment by mratsim** [comment]
I think the documentation should mention as a performance note that this needs a linear scan and is O(n).

**Comment by Araq** [comment]
But sets are of a bounded size in Nim so it's O(1). :P

**Comment by Wh1teDuke** [comment]
In my opinion 'sample' was an unneeded change.

**Comment by krux02** [comment]
I am ok with it.

File: lib/pure/random.nim [hunk 1]
```diff
@@ -171,6 +171,17 @@ proc rand*[T](a: openArray[T]): T {.deprecated.} =
   ## **Deprecated since v0.20.0:** use ``sample`` instead.
   result = a[rand(a.low..a.high)]
 
+proc sample*[T](r: var Rand; s: set[T]): T =
+  ## returns a random element from a set
+  var i = rand(r, card(s) - 1)
```
Comment (Araq) [inline lib/pure/random.nim]: Add least add an assertion that the set cannot be empty!

**Pull request merged**